tree/tree/src/TChain.cxx

```diff
@@ -268,87 +268,111 @@ Int_t TChain::Add(TChain* chain)
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-/// Add a new file to this chain.
+/// \brief Add a new file to this chain.
 ///
-/// Argument name may have either of two set of formats. The first:
-/// ~~~ {.cpp}
-///     [//machine]/path/file_name[?query[#tree_name]]
-///  or [//machine]/path/file_name.root[.oext][/tree_name]
+/// \param[in] name The path to the file to be added. See below for details.
+/// \param[in] nentries Number of entries in the file. This can be an estimate
+///            or queried from the file. See below for details.
+/// \returns There are different possible return values:
+/// - If nentries>0 (including the default of TTree::kMaxEntries) and no
+///   wildcarding is used, ALWAYS returns 1 irrespective of whether the file
+///   exists or contains the correct tree.
+/// - If wildcarding is used, regardless of the value of \p nentries, returns
+///   the number of files matching the name irrespective of whether they contain
+///   the correct tree.
+/// - If nentries<=0 and wildcarding is not used, returns 1 if the file
+///  exists and contains the correct tree and 0 otherwise.
+///
+/// \paragraph name_par Details of the name parameter
+/// There are two sets of formats accepted for the parameter \p name . The first
+/// one is:
+///
+/// ~~~{.cpp}
+///    [//machine]/path/file_name[?[query][#tree_name]]
+/// or [//machine]/path/file_name.root[.oext][/tree_name]
 /// ~~~
-/// If tree_name is missing the chain name will be assumed. Tagging the
-/// tree_name with a slash [/tree_name] is only supported for backward
-/// compatibility; it requires the file name to contain the string '.root'
-/// and its use is deprecated.
-/// Wildcard treatment is triggered by the any of the special characters []*?
-/// which may be used in the file name, eg. specifying "xxx*.root" adds
-/// all files starting with xxx in the current file system directory.
-///
-/// Alternatively name may have the format of a url, eg.
+///
+/// Note the following:
+/// - If the \p tree_name part is missing, it will be assumed that
+///   the file contains a tree with the same name as the chain.
+/// - The \p ?query part has two optional tokens: the "?" character and the
+///   query itself. A "?" with an empty query can be used as the opening token
+///   to give the name of the tree in the file (see the next point).
```

Comment:
i don't fully understand this sentence, but I have the impression that it can be removed given the information above and below it?

Reply:
I wanted to highlight the fact that both the "?" and the query part are separately optional, but the actual things that can be done with this are written in the bullets above and below, I agree

Reply:
they are not separately optional, you can't put `query` without the `?`

Reply:
Ok yes, only the opposite is possible. Is it worth mentioning explicitly or should I just remove this ?

Reply:
i think this can be removed